### PR TITLE
jormungandr: Update journey total duration at the end

### PR DESCRIPTION
**Pull request by kadhikari**
As fallback sections from origin and toward the destination are enriched only toward the end for selected journeys, we should also update total duration. All the other elements as listed below are correct.
* departure_date_time and arrival_date_time of the journey.
* departure_date_time, arrival_date_time and duration of all the sections are correct.
* The duration total per mode is correct.
* Update journey.duration and journey.durations.total

For details refer to: https://jira.kisio.org/browse/NAVITIAII-3155

Update co2_emission for journey adding co2_emission values of all section for each journey in the response.
For details refer to: https://jira.kisio.org/browse/NAVITIAII-3148

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 1]
```diff
@@ -345,6 +345,33 @@ def tag_journeys(resp):
...
+    """
+    update journey.co2_emission.value
+    """
+    if not pb_resp.journeys:
+        return
+    for j in pb_resp.journeys:
+        total_co2_emission = 0
+        for s in j.sections:
+            total_co2_emission += s.co2_emission.value
+        j.co2_emission.value = total_co2_emission
```
Comment (azime) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: minor:
replace : 
```python
total_co2_emission = 0
for s in j.sections:
    total_co2_emission += s.co2_emission.value
j.co2_emission.value = total_co2_emission
```
by: 
```python
j.co2_emission.value = sum(section.co2_emission.value for section in j.sections)
```

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 2]
```diff
@@ -345,6 +345,33 @@ def tag_journeys(resp):
...
+    """
+    if not pb_resp.journeys:
+        return
+    for j in pb_resp.journeys:
+        total_duration = 0
+        for s in j.sections:
+            total_duration += s.duration
+        j.durations.total = total_duration
+        j.duration = total_duration
+
```
Comment (azime) [inline source/jormungandr/jormungandr/scenarios/new_default.py:357]: minor:
replace : 
```python
total_duration = 0
for s in j.sections:
    total_duration += s.duration
j.durations.total = total_duration
j.duration = total_duration
```
by: 
```python
j.durations.total = sum(section.duration for section in j.sections)
j.duration = j.durations.total
```

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 1 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/scenarios/new_default.py]: :+1:

**Review by xlqian — APPROVED** [review]

**Review by azime — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 3]
```diff
@@ -345,6 +345,27 @@ def tag_journeys(resp):
...
 
+def update_durations(pb_resp):
+    """
+    update journey.duration and journey.durations.total
+    """
+    if not pb_resp.journeys:
+        return
+    for j in pb_resp.journeys:
+        j.duration = sum(s.duration for s in j.sections)
+        j.durations.total = j.duration
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/new_default.py:356]: Shouldn't we also recompute other things in `j.durations` like `walking/car/bike/etc` ?

**Review by pbench — APPROVED** [review]

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 3 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/scenarios/new_default.py:356]: Good point. As i never found any error on duration per mode i didn't do that.
:+1:

**Pull request merged**